web/src/pages/Login.jsx

```diff
@@ -1,10 +1,30 @@
 import { GithubFilled } from "@ant-design/icons";
 import { Flex, Typography } from "antd";
+import { useEffect, useContext } from "react";
+import { useNavigate } from "react-router-dom";
 
+import { AuthContext } from "../components/AuthContext";
 import { SubscriptionLogo } from "../components/SubscriptionLogo";
 import { ThemedButton } from "../components/ThemedButton";
 
+const CLIENT_ID = import.meta.env.VITE_GITHUB_CLIENT_ID;
+
 export default function Login() {
+	const navigate = useNavigate();
+	const { userData } = useContext(AuthContext);
+
+	useEffect(() => {
+		if (userData?.login) {
+			navigate("/dashboard");
+		}
+	}, [userData, navigate]);
+
+	const handleGitHubLogin = () => {
+		window.location.assign(
+			`https://github.com/login/oauth/authorize?client_id=${CLIENT_ID}`,
```

Comment:
The client_id is public (it is used in the url so everyone can see it) so it does not need to be kept secret and can just be entered in the code here